nlp/2021_07_07_neural_keyword_extraction/neural_keyword_extraction.ipynb

```diff
@@ -0,0 +1,1520 @@
+      },
+      "source": [
+        "## KeyBERT\n",
+        "\n",
+        "Strictly speaking [KeyBERT](https://github.com/MaartenGr/KeyBERT) is not an end-to-end neural keyword extraction model. \n",
+        "\n",
+        "Nontheless the **underlying idea** is as clever as it is simple: It compares embeddings of words with embeddings of texts and selects the set of keywords which are most similar to the entire text.\n",
+        "Both **word embeddings** as well was **text embeddings** are generated using **state-of-the-art neural models**, which have lead to tremendous performance improvements in other tasks.\n",
+        "\n",
+        "The benefit of this approach is that the used models are trained in an unsupervised manner and thus **do not require an annotated dataset** if keywords. In addition, they are available in a number of **non-English languages** as well.\n",
+        "\n",
+        "Let's see below how KeyBert can be used in practice."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "mV6d1GCs3sY6"
+      },
+      "source": [
+        "from keybert import KeyBERT"
+      ],
+      "execution_count": 2,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "3FdbT3Mi3xtY"
+      },
+      "source": [
+        "kw_model = KeyBERT()\n",
+        "\n",
+        "def extract_keybert_keywords(text, keyphrase_ngram_range=(1,1), stop_words='english',\n",
+        "                             use_maxsum=True, nr_candidates=20, top_n=5,\n",
+        "                             use_mmr=False, diversity=0.7,\n",
+        "                             only_keywords=True):\n",
+        "    keyphrases = kw_model.extract_keywords(text, keyphrase_ngram_range=keyphrase_ngram_range, \n",
+        "                                           stop_words=stop_words, use_maxsum=use_maxsum, \n",
+        "                                           use_mmr=use_mmr, diversity=diversity, \n",
+        "                                           nr_candidates=nr_candidates, top_n=top_n)\n",
+        "    if only_keywords:\n",
+        "        keyphrases = [phrase for phrase, score in keyphrases]\n",
+        "    return keyphrases"
+      ],
+      "execution_count": 62,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "dmYw5b02sUFq"
+      },
+      "source": [
+        "## BART-based model\n",
+        "\n",
+        "🤗 Hugging Face Model Hub is *the* prime address when it comes to discovering state-of-the-art models that are easy to use. However, with respect to neural keyword (or keyphrase) extraction there are only two models available as of this writing.\n",
+        "\n",
+        "Both models were created by [Ankur Singh](https://huggingface.co/ankur310794) and use a **BART-based sequence-to-sequence architecture**. Unfortunately is not a lot of details available about the training specific \n",
```

Comment:
Bit of a strange sentence: `Unfortunately is not a lot of details available about the training specific` > `Unfortunately there are not a lot of details available about the training specifics`